akamai/resource_akamai_fastdns_zone.go

```diff
@@ -1287,3 +1288,26 @@ func resourceFastDNSZoneExists(d *schema.ResourceData, meta interface{}) (bool,
 	zone, err := dns.GetZone(hostname)
 	return zone != nil, err
 }
+
+func saveZone(zone *dns.Zone) error {
+	log.Printf("[DEBUG] [Akamai FastDNS] Saving zone")
+	if zone.Zone.Soa != nil {
+		zone.Zone.Soa.Serial += 1
+	}
+	return zone.Save()
+}
+
+// 'serial' and 'minimum' were changed from int to uint in v0.5.2 of the Akamai
+// client. Terraform schema doesn't support uint, so we use these functions to
+// convert during marshaling. The program will panic on overflow. This is
+// unlikely in practice because 'minimum' tends to be low, and 'serial' starts
+// as a UNIX time (much lower than max int) and increases by one when saved.
+func convertSoaUintsToInts(soa map[string]interface{}) {
```

Comment:
if I recall correctly this was an issue in more record types than just SOA.  See https://github.com/akamai/terraform-provider-akamai/issues/34

Sadly the test coverage in this plugin is very poor. If we are going to change the underlying API it might be worth writing more extensive tests if possible.

Reply:
Yes, I see int to uint changes in SRV and NAPTR as well. Wasn't sure if these were relevant to our scenarios.

I wonder about investing time in writing tests for this library. It might be a better use of time to write our own, or we could just remove Akamai from Terraform. We can discuss out of band.